x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -26,13 +33,15 @@
  * These functions needs to be pushed down to Lucene queries to be executed - there's no Evaluator for them, but depend on
  * {@link org.elasticsearch.xpack.esql.optimizer.LocalPhysicalPlanOptimizer} to rewrite them into Lucene queries.
  */
-public abstract class FullTextFunction extends Function {
+public abstract class FullTextFunction extends Function implements TranslationAware {
 
     private final Expression query;
+    private final QueryBuilder queryBuilder;
 
-    protected FullTextFunction(Source source, Expression query, List<Expression> children) {
+    protected FullTextFunction(Source source, Expression query, List<Expression> children, QueryBuilder queryBuilder) {
         super(source, children);
         this.query = query;
+        this.queryBuilder = queryBuilder;
```

Comment:
`QueryBuilder` is not a `child` of `FullTextFunction`, perhaps there is no need to add it to its children list if the goal is letting `ES|QL` planner treat `QueryBuilder` as a blackbox. I couldn't think of a case that could cause troubles for now, just to keep a note here.

Reply:
@fang-xing-esql I don't understand this comment. queryBuilder is already not added to the children list. What am I missing?

Reply:
I understand `QueryBuilder` is not in the children list, I was wondering whether it needs to be added as a child, I couldn't think of a good reason to add it yet.